mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -1167,7 +1167,19 @@ bool GetCheckedIPs (SocketAsyncEventArgs e, out IPAddress [] addresses)
 			// while skipping entries that do not match the address family
 			DnsEndPoint dep = e.RemoteEndPoint as DnsEndPoint;
 			if (dep != null) {
-				addresses = Dns.GetHostAddresses (dep.Host);
+				var possibleAddresses = Dns.GetHostAddresses (dep.Host);
```

Comment:
I don't think we need to allocate 2 additional arrays. It should be enough to shuffle the items in the original array and resize it at the end (if necessary)

Reply:
I agree that would improve performance, but I think it would make the code more difficult to understand. I'd lean toward easier to understand code here, as I don't see this as being performance critical.

Reply:
I don't this so. Typing blindly but following code should do same

```csharp
		var possibleAddresses = Dns.GetHostAddresses (dep.Host);
		int last_valid = 0;
		for (int i = 0; i < possibleAddresses.Length; ++i) {
			if (possibleAddresses [i].AddressFamily != dep.AddressFamily)
				continue;

			possibleAddresses [last_valid++] = possibleAddresses [i];
		}

		if (last_valid != possibleAddresses.Length)
			Array.Resize (ref possibleAddresses, last_valid);
```

and it looks to me to be easier to understand

Reply:
Ok, I'll go with a change similar to this one. Thanks!